### ISSUE TITLE
关于损失函数的定义

### ISSUE THREAD

**Issue by xiajinbiao** [Tianxiaomo/pytorch-YOLOv4#170]
请问在train.py下的Yolo_loss里面，在bounding box regression的时候，为什么没有用到CIoU呢，只是在筛选anchor的时候用到。此外，在tool/region_loss.py里面使用MSE进行bounding box regression。论文中不是提到用CIoU-loss吗？是因为作者公开的源码和论文内容不全都一致吗？（新手入门，如有不对，还请指教）

Yolo_loss部分：
            loss_xy += F.binary_cross_entropy(input=output[..., :2], target=target[..., :2],
                                              weight=tgt_scale * tgt_scale, size_average=False)
            loss_wh += F.mse_loss(input=output[..., 2:4], target=target[..., 2:4], size_average=False) / 2
            loss_obj += F.binary_cross_entropy(input=output[..., 4], target=target[..., 4], size_average=False)
            loss_cls += F.binary_cross_entropy(input=output[..., 5:], target=target[..., 5:], size_average=False)
            loss_l2 += F.mse_loss(input=output, target=target, size_average=False)

RegionLoss部分：
           loss_x = self.coord_scale * nn.MSELoss(size_average=False)(x * coord_mask, tx * coord_mask) / 2.0
           loss_y = self.coord_scale * nn.MSELoss(size_average=False)(y * coord_mask, ty * coord_mask) / 2.0
           loss_w = self.coord_scale * nn.MSELoss(size_average=False)(w * coord_mask, tw * coord_mask) / 2.0
           loss_h = self.coord_scale * nn.MSELoss(size_average=False)(h * coord_mask, th * coord_mask) / 2.0
           loss_conf = nn.MSELoss(size_average=False)(conf * conf_mask, tconf * conf_mask) / 2.0
           loss_cls = self.class_scale * nn.CrossEntropyLoss(size_average=False)(cls, tcls)
           loss = loss_x + loss_y + loss_w + loss_h + loss_conf + loss_cls

**Comment by Tianxiaomo** [comment]
https://github.com/AlexeyAB/darknet/blob/fe4b2d008e7aa168e031400e531d4b462ee9d326/src/yolo_layer.c#L636-L640

可能我理解不对吧
